src/__tests__/lib/containers/table/datasets/DatasetItemsEditor.test.tsx

```diff
@@ -0,0 +1,564 @@
+    await removeItem('syn999')
+
+    await screen.findByText('1 File', { exact: true })
+  })
+
+  describe('Remove Items', () => {
+    it('Disables the remove button until an item has been selected', async () => {
+      // Start with one item
+      const getDatasetHandler = getDatasetHandlerWithItems([
+        referenceToDatasetItem(mockFileReference),
+      ])
+      server.use(getDatasetHandler)
+
+      await renderComponent()
+
+      // The Remove button should be disabled.
+      await waitFor(() =>
+        expect(
+          screen.getByRole('button', {
+            exact: false,
+            name: 'Remove Items',
+          }),
+        ).toBeDisabled(),
+      )
+
+      await selectIndividualItem(mockFileReference.targetId)
+
+      // The remove item should not be disabled after making a selection
+      await waitFor(() =>
+        expect(
+          screen.getByRole('button', {
+            exact: false,
+            name: 'Remove Items',
+          }),
+        ).not.toBeDisabled(),
+      )
+    })
+
+    it('Removes selected items when the button is clicked', async () => {
+      // Start with one item
+      const getDatasetHandler = getDatasetHandlerWithItems([
+        referenceToDatasetItem(mockFileReference),
+      ])
+      server.use(getDatasetHandler, successfulUpdateHandler)
+
+      await renderComponent()
+
+      await removeItem(mockFileReference.targetId)
+
+      await clickSave()
+
+      // Verify that there are no dataset items
+      await waitFor(() =>
+        expect(updatedEntityCaptor).toBeCalledWith(
+          expect.objectContaining({ items: [] }),
+        ),
+      )
+
+      // Since we removed the selected item, there should now be nothing selected
+      await verifyNoneSelected()
+    })
+  })
+
+  it('Allows selecting versions and updates the selected version when a new one is picked', async () => {
+    const getDatasetHandler = getDatasetHandlerWithItems(
+      [mockFileReference].map(referenceToDatasetItem),
+    )
+    server.use(getDatasetHandler, successfulUpdateHandler)
+    await renderComponent()
+
+    // Sanity check: the selected version should not be 1 when we start.
+    expect(mockFileReference.targetVersionNumber).not.toEqual(1)
+
+    userEvent.selectOptions(await screen.findByRole('listbox'), '1')
+
+    await clickSave()
+
+    // Verify that the item has changed
+    await waitFor(() =>
+      expect(updatedEntityCaptor).toBeCalledWith(
+        expect.objectContaining({
+          items: [
+            referenceToDatasetItem({
+              targetId: mockFileEntity.id!,
+              targetVersionNumber: 1,
+            }),
+          ],
+        }),
+      ),
+    )
+  })
+
+  /**
+   * TODO: This test passes on its own but fails in the suite.
+   * We're calling resetHandlers and clearAllMocks, so I have no idea why it isn't working.
+   */
+  it.skip('Handles an error on Save by displaying a toast and not calling the callback', async () => {
```

Comment:
I spent a ton of time trying to figure this out...the test works on its own and the behavior is correct when I take these steps in the browser. It just fails in the test suite. Mocks are reset after each test, so I have no idea why this isn't working.

Reply:
This is unfortunate, and something I've also encountered and can't explain.  @conundrumgirl : Any ideas?